src/lib/events/gtmDlPushEvent.js

```diff
@@ -24,12 +26,23 @@ const handler = function (argEvent) {
     const eventModel =
       argEvent && argEvent.detail && argEvent.detail.eventModel;
 
+    if (doConvertArrayEvents && eventModel) {
```

Comment:
@rfreemo , the implemented code works fine. But if you want to refine it a bit, please check my comment below.

As the intention is to check whether eventModel is an array, eventModel[0] === "event" and eventModel[1] exists, it can be simplified as below

if (doConvertArrayEvents && Array.isArray(eventModel) &&  eventModel[0] === constants.KEYWORD_EVENT && eventModel[1]) {
  eventModel = {
    event: eventModel[1],
    gaArrayEvent: eventModel[2] ? eventModel[2] : undefined
  }
}

It will also completely replace the eventModel array with an Object containing event and gaArrayEvent properties.

Reply:
@arcreddy the logic here is slightly different - !eventModel.event is not included in the proposal and arrayEvent is a specific type of array rather than an array test.  I would prefer to leave it in the slightly verbose two step process, especially as I have been testing with it.

This does raise a major issue that the implementation is fragile and not easy to understand despite my efforts to keep the code clean and variables well named.  What we actually need is good tests, but that's quite a significant task - for me at least

Reply:
@brobatr  sure, that make sense.